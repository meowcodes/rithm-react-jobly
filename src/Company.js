import React, { Component } from 'react';
import JoblyApi from './JoblyAPI';
import JobCard from './JobCard';

/**
 * Recieves ONE company data from API request
 * Renders ONE company details
 */
class Company extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: true,
      companyData: {},
      error: null
    }
  }

  // gets company data and adds it to state. changes
  // loading to false
  async componentDidMount() {
    try{
      let data = await JoblyApi.getCompany(this.props.handle);
      this.setState({
        companyData: data,
        loading: false
      });
    } catch(err) {
      this.setState({
        error: err
      });
    }
  }

  render() {
    // set of job ids of jobs user has applied to
    const appliedJobs = new Set(this.props.currJobs.map( job => job.id ));
    const company = this.state.companyData;
    let jobs;
  
    if (!this.state.loading) {
      jobs = this.state.companyData.jobs.map( job => 
        <JobCard 
          title={job.title} 
          salary={job.salary} 
          equity={job.equity}
          key={ job.id }
          applied={ appliedJobs.has(job.id) }
          triggerApply={ () => this.props.triggerApply(job.id) }
        />
      )
    }

    return (
      <div className="Company">
        { this.state.error && this.state.loading
          ? <p>Loading...</p>
          : <div>
              <p className="Company-name">
                { company.name }
              </p>

              <p className="Company-desc">
                { company.description }
              </p>

              <div className="Company-jobs">
                {jobs}
              </div>
          </div>
        } 
        { this.state.error && <p>Something went wrong</p>}
      </div>
    );
  }
}

export default Company;
