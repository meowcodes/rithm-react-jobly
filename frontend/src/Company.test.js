import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { MemoryRouter } from 'react-router-dom';
import Company from './Company';


// full mount
it('mounts without crashing', function() {
  mount(
    <MemoryRouter>
      <Company 
        handle="test"
        currJobs={[]}
        triggerApply={()=>null}/>
    </MemoryRouter>
  );
});

// snapshot test
it('matches snapshot', function() {
  let wrapper = shallow(<Company 
                          handle="test"
                          currJobs={[]}
                          triggerApply={()=>null}/>);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
