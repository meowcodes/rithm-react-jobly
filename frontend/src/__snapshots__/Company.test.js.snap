// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Company"
>
  <div>
    <p
      className="Company-name"
    />
    <p
      className="Company-desc"
    />
    <div
      className="Company-jobs"
    />
  </div>
</div>
`;
